.github/workflows/_build_docker.yaml

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2022 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+# This reuseable workflow builds the Macaron Docker image.
+# If release-tag is provided. The final image will be tagged and push to ghcr.io/oracle-samples/macaron:release-tag
+
+name: Build and push Docker image
+on:
+  workflow_call:
+    inputs:
+      release-tag:
+        description: The release tag to push the Docker image
+        required: false
+        type: string
+
+env:
+  IMAGE_NAME: ghcr.io/oracle-samples/macaron
+
+jobs:
+  build-docker:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+      with:
+        fetch-depth: 0
+
+    - name: Download artifact
+      uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
+      with:
+        path: dist
+
+    - name: Log in to GitHub Container Registry
+      run: docker login ghcr.io --username ${{ github.actor }} --password ${{ secrets.GITHUB_TOKEN }}
```

Comment:
If the workflow is called from the `pr-change-set.yaml`, I don't understand why we would need to login because we won't push the image to the registry.

Reply:
Currently, the image `ghcr.io/oracle-samples/macaron-base` is private. The login step is needed so that we could pull it for building the final Macaron Docker image.

Reply:
We need to make sure the workflows that are triggered with `pull_request` event have the minimal permissions and accesses because it can contain untrusted code. So we should not allow docker login for pull requests.